lib/clients/statsd.js

```diff
@@ -0,0 +1,86 @@
+var BaseClient = require('./base');
+var util = require('util');
+var types = require('../types');
+var _ = require('lodash');
+var dgram = require('dgram');
+
+var STATS_TYPES = {};
+STATS_TYPES[types.C] = 'c';
+STATS_TYPES[types.G] = 'g';
+STATS_TYPES[types.T] = 'ms';
+
+var KEY_DELIMITER = "-";
+
+function sanatizeInput(input) {
+  //Replace characters that may cause problems here (:| are used by statsd, and cannot be included in the name)
+  return (input + '').replace(/[\s:|=<.]+/g, '');
+}
+
+//a default implementation to format the meta data into the name of the metric
+//We will embed tags into the key. Each message will be constructed like this:
+// <key>-<tagKey>_is_<tagValue>...-<fieldName>:<fieldValue>|<type>
+// For example, given this data:
+// {type: 'gauge', key: 'memory', tags: {host: localhost, workerId:1}, fields: {used: 800, total: 1000}}
+// it will be converted to:
+// ['memory-host_is_localhost-workerId_is_1-used:800|g', 'memory-host_is_localhost-workerId_is_1-total:1000|g']
+function defaultMetricsKeyBuilder(keyField, tagsFields, fieldName) {
+  var keys = [];
+  //the key field first
+  if (keyField) {
+    keys.push(keyField);
+  }
+  //then tags. statsd doesn't really support tags, so we just add them as part of the key.
+  if (tagsFields) {
+    var tags = _.map(tagsFields, function(tagValue, tagKey) {
+      return tagKey + '_is_' + tagValue; //see https://github.com/vimeo/carbon-tagger
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Move this trailing comment on the previous empty line. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingComment)